plugins/woocommerce-blocks/assets/js/blocks/single-product/edit/index.tsx

```diff
@@ -53,6 +53,10 @@ const Editor = ( {
 	const [ isEditing, setIsEditing ] = useState( ! productId );
 	const blockProps = useBlockProps();
 
+	const block = useSelect( ( select ) =>
+		select( 'core/blocks' ).getBlockType( metadata.name )
+	);
```

Comment:
```suggestion
	const block = useSelect( ( select ) =>
		select( 'core/blocks' ).getBlockType( metadata.name ), [ ]);
```

Let's pass an empty array as a dependency array, so `getBlockType` is invoked only one time.

Reply:
Done, thanks!